### PR TITLE
Guozhong/improve auto infer request line coverage

### PR DESCRIPTION
**Pull request by wgzintel**
### Details:
 - *Improve line coverage of auto_infer_request.cpp*
 - *Encapsulate getNetwork() in the base class InferRequestVariableStateTest, which can be used by CPU, GPU, and GNA VariableStateTest*

### Tickets:
 - *ticket-99028*

### CODE REVIEW

File: src/plugins/auto/infer_request.cpp [hunk 1]
```diff
@@ -91,13 +91,6 @@ void MultiDeviceInferRequest::SetBlob(const std::string& name, const InferenceEn
         IInferRequestInternal::SetBlob(name, blob);
 }
 
-void MultiDeviceInferRequest::SetBlob(const std::string& name, const Blob::Ptr& blob, const PreProcessInfo& info) {
```
Comment (wgzintel) [inline src/plugins/auto/infer_request.cpp:94]: Can we remove SetBlob according to https://github.com/openvinotoolkit/openvino/blob/master/src/inference/include/ie/cpp/ie_infer_request.hpp Line 117

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_querystate.cpp [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        using namespace CommonTestUtils;
+        for (auto& configItem : configuration) {
+            result << "configItem=" << configItem.first << "_";
+            configItem.second.print(result);
+        }
+    }
+    return result.str();
+}
+
+TEST_P(OVInferRequestQueryStateTest, QuerystateTest) {
```
Comment (songbell) [inline src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_querystate.cpp:27]: please double check if any case overlap or integration can be made with below:
https://github.com/openvinotoolkit/openvino/pull/11590/files#diff-e8c107eadd83bad9ed3c25790b3f8682abc7b3b216977e9883119caa207b104c

File: src/plugins/auto/infer_request.cpp (hunk 1 above)
Comment (songbell) [reply src/plugins/auto/infer_request.cpp:94]: https://docs.openvino.ai/latest/openvino_docs_OV_UG_Preprocessing_Overview.html
need to under stand preprocess steps

Comment (wgzintel) [reply src/plugins/auto/infer_request.cpp:94]: Updated

File: src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_querystate.cpp (hunk 2 above)
Comment (wgzintel) [reply src/tests/functional/plugin/shared/src/behavior/ov_infer_request/infer_querystate.cpp]: Updated

File: src/plugins/auto/plugin.cpp [hunk 3]
```diff
@@ -956,7 +956,7 @@ std::vector<DeviceInformation> MultiDeviceInferencePlugin::FilterDeviceByNetwork
     });
 
     // If CPU is in candidate list, load dynamic network to CPU first
-    if ((model->is_dynamic() || isStateful()) && cpuiter != metaDevices.end()) {
+    if ((model->is_dynamic() || (isStateful() && !isMulti)) && cpuiter != metaDevices.end()) {
```
Comment (wangleis) [inline src/plugins/auto/plugin.cpp:959]: Please check if AUTO folder has only one MULTI flag. If no, please update to single one.

Comment (wgzintel) [reply src/plugins/auto/plugin.cpp:959]: Create JIRA: https://jira.devtools.intel.com/browse/CVS-103669 to follow up.

File: src/plugins/auto/infer_request.cpp [hunk 4]
```diff
@@ -91,12 +91,14 @@ void MultiDeviceInferRequest::SetBlob(const std::string& name, const InferenceEn
         IInferRequestInternal::SetBlob(name, blob);
 }
 
+IE_SUPPRESS_DEPRECATED_START
```
[the remark this answers is not in the document]
Comment (wgzintel) [reply src/plugins/auto/infer_request.cpp:94]: Confirm with xuejun, no additional MACRO is required.

File: src/plugins/intel_gna/tests/functional/shared_tests_instances/behavior/infer_request/memory_states.cpp [hunk 5]
```diff
@@ -12,39 +12,10 @@
...
-    sigm->add_control_dependency(mem_w2);
-
-    auto function =
-        std::make_shared<ngraph::Function>(ngraph::NodeVector{sigm}, ngraph::ParameterVector{input}, "addOutput");
-    return InferenceEngine::CNNNetwork{function};
-}
-
-std::vector<memoryStateParams> memoryStateTestCases = {
-    memoryStateParams(getNetwork(), {"c_1-3", "r_1-3"}, CommonTestUtils::DEVICE_GNA, {})};
+std::vector<memoryStateParams> memoryStateTestCases = {memoryStateParams(InferRequestVariableStateTest::getNetwork(),
```
Comment (wangleis) [inline src/plugins/intel_gna/tests/functional/shared_tests_instances/behavior/infer_request/memory_states.cpp:15]: why we need test case in GNA for AUTO?

Comment (wgzintel) [reply src/plugins/intel_gna/tests/functional/shared_tests_instances/behavior/infer_request/memory_states.cpp:15]: GNA already has this test, currently only encapsulate getNetwork() in the base class InferRequestVariableStateTest, which can be used by CPU, GPU, and GNA VariableStateTest.

**Review by wangleis — APPROVED** [review]

**Pull request merged**